jdbc/src/main/java/org/apache/metamodel/jdbc/dialects/HiveQueryRewriter.java

```diff
@@ -35,6 +35,10 @@ public String rewriteColumnType(ColumnType columnType, Integer columnSize) {
         if (columnType == ColumnType.INTEGER) {
             return "INT";
         }
+        // Hive does not support VARCHAR without a width, nor VARCHAR(MAX).
+        if (columnType == ColumnType.STRING && columnSize == null) {
```

Comment:
Ah yes. Will fix. I'll also look for other VARCHAR "convertees" (I'm not actually sure if other non-VARCHAR types are in danger of the same issue... I'll see if I can find anything).